menu/src/main/java/com/jagrosh/jdautilities/menu/MessagePaginator.java

```diff
@@ -0,0 +1,579 @@
+            case LEFT:
+            case STOP:
+            case RIGHT:
+                return isValidUser(event.getUser(), event.getGuild());
+            case BIG_LEFT:
+            case BIG_RIGHT:
+                return bulkSkipNumber > 1 && isValidUser(event.getUser(), event.getGuild());
+            default:
+                return false;
+        }
+    }
+
+    // Private method that handles MessageReactionAddEvents
+    private void handleMessageReactionAddAction(MessageReactionAddEvent event, Message message, int pageNum)
+    {
+        int newPageNum = pageNum;
+        switch(event.getReaction().getReactionEmote().getName())
+        {
+            case LEFT:
+                if(newPageNum == 1 && wrapPageEnds)
+                    newPageNum = pages + 1;
+                if(newPageNum > 1)
+                    newPageNum--;
+                break;
+            case RIGHT:
+                if(newPageNum == pages && wrapPageEnds)
+                    newPageNum = 0;
+                if(newPageNum < pages)
+                    newPageNum++;
+                break;
+            case BIG_LEFT:
+                if(newPageNum > 1 || wrapPageEnds)
+                {
+                    for(int i = 1; (newPageNum > 1 || wrapPageEnds) && i < bulkSkipNumber; i++)
+                    {
+                        if(newPageNum == 1 && wrapPageEnds)
+                            newPageNum = pages + 1;
+                        newPageNum--;
+                    }
+                }
+                break;
+            case BIG_RIGHT:
+                if(newPageNum < pages || wrapPageEnds)
+                {
+                    for(int i = 1; (newPageNum < pages || wrapPageEnds) && i < bulkSkipNumber; i++)
+                    {
+                        if(newPageNum == pages && wrapPageEnds)
+                            newPageNum = 0;
+                        newPageNum++;
+                    }
+                }
+                break;
+            case STOP:
+                finalAction.accept(message);
+                return;
+        }
+
+        try {
+            event.getReaction().removeReaction(event.getUser()).queue();
+        } catch(PermissionException ignored) {}
+
+        int n = newPageNum;
+        message.editMessage(renderPage(newPageNum)).queue(m -> pagination(m, n));
+    }
+    
+    private Message renderPage(int pageNum)
+    {
+        return messageFunction.apply(pageNum, pages);
+    }
+
+    /**
+     * The {@link com.jagrosh.jdautilities.menu.Menu.Builder Menu.Builder} for
+     * a {@link com.jagrosh.jdautilities.menu.MessagePaginator MessagePaginator}.
+     *
+     * @author John Grosh
+     * @author Adrian Todt
+     */
+    public static class Builder extends Menu.Builder<Builder, MessagePaginator>
+    {
+        private BiFunction<Integer,Integer,Message> function = (page, pages) -> null;
+        private Consumer<Message> finalAction = m -> m.delete().queue();
+        private int items = 1;
```

Comment:
This is actually the amount of pages as confirmed by variable name and usage of the actual Menu implementation. Therefore renaming this variable would be wise to reflect that.